src/main/java/io/vertx/ext/stomp/utils/Server.java

```diff
@@ -0,0 +1,30 @@
+package io.vertx.ext.stomp.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Scanner;
+
+/**
+ * Class responsible for the computation of the server id. From the STOMP specification, this id must be constructed as
+ * follows: name/version comments.
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+public class Server {
+
+  public static final String SERVER_NAME;
+
+  static {
+    try (InputStream is = Server.class.getClassLoader().getResourceAsStream("vertx-stomp-version.txt")) {
```

Comment:
That's the vertx-stomp jar version, not the set of supported protocols (configurable in the options). You cannot have two versions of the jar in a single JVM using a flat classloader (without funky bugs). You would need to separate the two class spaces, and so static would have different values.